crates/core/src/client.rs

```diff
@@ -290,3 +292,37 @@ async fn sn_state_block_number_inner(
 
     Ok(U256::from_big_endian(&sn_block_num).as_u64())
 }
+
+async fn wrapped_retry<'a, Fut, T, E, D>(
```

Comment:
Thanks for the time reviewing. 

This crate does not timeout the async function (as mentioned in #561 ) and expects a simple`Result<T>` for retrying. However, when timing the async function (i.e., `future::timeout(duration, async_fn())`), it will return Result<Result<T, CodeError>, TimeoutError>. Thus, when the function is executed in time but raises `CodeError`, additional handling is required. Consequently, considering implementing our custom retry function is not hard, I think we don't need to import and use this crate. What do you think?

Reply:
I will resolve this conversation as a new implementation is pushed that removes retrying. Feel free to reopen this if necessary.